cmd/gateway-azure.go

```diff
@@ -621,11 +621,7 @@ func (a *azureObjects) PutObjectPart(bucket, object, uploadID string, partID int
 		return info, err
 	}
 
-	etag := data.md5Sum
-	if etag == "" {
-		// Generate random ETag.
-		etag = getMD5Hash([]byte(mustGetUUID()))
-	}
+	etag := azureToS3ETag(getMD5Hash([]byte(mustGetUUID())))
```

Comment:
You would need to keep `etag := data.md5Sum`

Reply:
In this case, I should keep the `azureToS3ETag(etag)` on line 654 `info.ETag = azureToS3ETag(etag)` to be sure to have a suffixed etag, no?

Reply:
Or just, after the `if` add `etag = azureToS3ETag(etag)` ?

Reply:
below is the change required
```go
	etag := data.md5Sum
	if etag == "" {
		// Generate random ETag.
		etag = azureToS3Tag(getMD5Hash([]byte(mustGetUUID())))
	}
```

Reply:
It will no work because because some time you will have an etag containing a real md5Sum without suffix.

Reply:
@jmaitrehenry Yep. Unable to understand why it doesn't work.

Reply:
@balamurugana Finally, I made a mistake when I double check it.
I made the change asked and it works. Sorry for that :/